app/models/AFTSubmitterDetails.scala

```diff
@@ -16,14 +16,27 @@
 
 package models
 
-import play.api.libs.json.{Format, Json}
+import play.api.libs.json.{Format, Json, Reads}
 
-import java.time.LocalDate
+import java.time.{LocalDate, LocalDateTime}
 
 case class AFTSubmitterDetails(submitterType: String, submitterName: String, submitterID: String, authorisingPsaId: Option[String], receiptDate: LocalDate)
 
 object AFTSubmitterDetails {
   implicit val formats: Format[AFTSubmitterDetails] = Json.format[AFTSubmitterDetails]
+
+  implicit val readAftDetailsFromIF: Reads[AFTSubmitterDetails] = Reads { json =>
+    for {
+      submitterType <- (json \ "aftDeclarationDetails" \ "submittedBy").validate[String]
+      submitterName <- (json \ "aftDeclarationDetails" \ "submitterName").validate[String]
+      submitterID <- (json \ "aftDeclarationDetails" \ "submitterId").validate[String]
+      authorisingPsaId = (json \ "aftDeclarationDetails" \ "psaId").asOpt[String]
+      receiptDate <- (json \ "aftDetails" \ "receiptDate").validate[String].map { dateTime =>
+        LocalDateTime.parse(dateTime.dropRight(1)).toLocalDate
```

Comment:
This is copying how they did it: https://github.com/hmrc/pension-scheme-accounting-for-tax/blob/main/app/transformations/ETMPToUserAnswers/AFTDetailsTransformer.scala#L82-L89